### PR TITLE
feat: update cypress version and config

### LINKED ISSUE #39003: Cypress failing to install during ci
For example:
https://github.com/freeCodeCamp/freeCodeCamp/runs/737483891

While Cypress installation is intended to be skipped locally, https://github.com/freeCodeCamp/freeCodeCamp/pull/38660, it needs to be installed for ci.  It's possible this is a caching issue, but we might simply need to provide `CYPRESS_INSTALL_BINARY` during ci.  Hopefully it's the former, since we don't want to have to keep updating the environment variables every time we update Cypress.

### PR DESCRIPTION
**Pull request by raisedadead**
- [x] Bumps the cypress version (and some other things, too)
- [x] Re-organises the scripts a little
- [ ] ~~Move to `start-server-and-test` instead of the shell script ref: https://github.com/freeCodeCamp/freeCodeCamp/issues/38679~~
- [ ] ~~Update docs~~

Closes #39003

### CODE REVIEW

**Comment by raisedadead** [comment]
The Cypress action comes with batteries included. 

They include `npm install (via npm ci) and some aggressive caching maybe to prevent straining their backends. We have co-incidentally blocked the Cypress binary installation using a `.npmrc`. This makes the binary unavailable explicitly in the CI, aagh!

Moving on our on-demand installation does not work either because of the way Github actions cache node_modules and cache directories.

Since Cypress is needed to install cypress, which is not available its a vicious loop.

Long story short, this was working because of a cached binary and a major bump + our explicit enforcement of not installing the binary breaks things.

The fix is to set the env variable to the version that is expected.

**Comment by raisedadead** [comment]
It might be a while to get around to doing these two:

> * [ ] Move to `start-server-and-test` instead of the shell script ref: #38679
> * [ ] Update docs

I am opening this up for review to address the docs, and the better tooling later.
